doc/tutorials/java-plugins.md

```diff
@@ -0,0 +1,55 @@
+# How-To: Write a Java Plugin
+
+This file serves as a tutorial on how to get started with writing a java plugin.
+
+## Basics
+
+If you want to know more about plugins in general, please check out the [How-To: Write a Plugin](/doc/tutorials/plugins.md) page.
+If you need a tutorial for java, please check out the [How-To: Java kdb](/doc/tutorials/java-kdb.md) page.
+
+## Two types of Java Plugins
+
+Before we will take a look into how to write a plugin in Java, it is important to note, that there are two ways of doing that:
+
+- using JNI plugin
+- using JNA binding
+
+If you would like to write a plugin in Java and look into the documentation, then you will definitely meet two notions like `JNI Plugin` and `JNA Binding`, so let's clarify, what they are.
+
+### JNI Plugin
+
+Java Native Interface is framework, which allows the Java code to execute or to be executed by the programms, written in other languages like C, C++ and Assembly. Since most of Elektra plugins are written in C and C++, the developer can use JNI plugin, which was created specifically for Elektra. This plugin allows developers to integrate Java into C code. More on JNI Plugins, you can read [here](/src/plugins/jni/README.md).
+
+In order to use `JNI` in Elektra, the library `jni.h` must be provided for it. It is important to note, that Elektra uses SDK version 1.8, so you have to install this version of SDK and use while writing a Java plugin. This file is used later in `JNI Plugin`, which has to be enabled, while installing Elektra. More information on how to install `JNI Plugin` can be found [here](/src/plugins/jni/README.md). Please read carefully and follow instructions in `Installation` and `Enabling the Plugin` sections. If you don't have `jni.h` file, it can be easily downloaded and placed under this path:
```

Comment:
I think the JNI plugin should also work with newer versions of Java. At least the [Debian unstable build job includes the plugin](https://build.libelektra.org/jenkins/blue/organizations/jenkins/libelektra/detail/master/764/pipeline/420/#step-648-log-92) after [your PR](https://github.com/ElektraInitiative/libelektra/pull/2784) was merged.

```suggestion
In order to use `jni` in Elektra, the library `jni.h` must be provided for it. This file is used later in “JNI Plugin”, which has to be enabled, while installing Elektra. More information on how to install the “JNI Plugin” can be found [here](/src/plugins/jni/README.md). Please read carefully and follow instructions in the “Installation” and “Enabling the Plugin” sections. If you don't have the `jni.h` file, it can be easily downloaded and placed under this path:
```